src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -135,7 +158,14 @@ public boolean rename(Path src, Path dst) throws IOException {
      */
     @Override
     public boolean delete(Path f, boolean recursive) throws IOException {
-        return false;
+        FileSystemPathTuple tuple = generateFSPathTuple(f);
+        /*
+        Potential Bug:
+            If `recursive == true`, the recursive process may cause the underlying filesystems to change.
+            For example, if "s3a://bucket/dir1/" is mapped to "fs1://bla/", and "s3a://bucket/dir1/dir2/" is mapped to
+            "fs2://blah/" then the recursive process will fail if the path points to "s3a://bucket/" or "s3a://bucket/dir1/"
```

Comment:
Thanks for documenting this. That was one of my concerns. did you test it with our Sonatype Spark app that (I think) is doing some deletes during the write process?

Reply:
Nope. Not aware of this spark app... What is it?

Reply:
https://github.com/treeverse/lakeFS/tree/master/test/spark/app